internal/apierror/api_error.go

```diff
@@ -16,6 +16,11 @@ import (
 //
 // APIErrorInterface should be used with errors.As instead of this struct.
 type APIError struct {
+	// InternalError is an additional error that might be associated with the
+	// API error. It's not returned in the API error response, but is logged in
+	// API endpoint execution to provide extra information for operators.
+	InternalError error `json:"-"`
```

Comment:
Added a possible internal error that an API error can be tagged with and which will be added to the logs by `api_endpoint.go` above. This is nice because it lets some additional context be added for operators without the possibility of leaking anything sensitive from an internal error via HTTP.

Reply:
Love it!